nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -392,24 +418,33 @@ def _test_binary(op, arrseq, scalars, seqs, inplace=False):
                 continue  # Going to deal with it separately.
 
             _test_binary(op, seq_int, [42, -3, True, 0], [seq_int, seq_bool, -seq_int], inplace=True)  # int <-- int
-            assert_raises(TypeError, _test_binary, op, seq_int, [0.5], [], inplace=True)  # int <-- float
-            assert_raises(TypeError, _test_binary, op, seq_int, [], [seq], inplace=True)  # int <-- float
+            
+            with pytest.raises(TypeError):
+                _test_binary(op, seq_int, [0.5], [], inplace=True) # int <-- float
+                _test_binary(op, seq_int, [], [seq], inplace=True) # int <-- float
+            
 
         # __pow__ : Integers to negative integer powers are not allowed.
         _test_binary("__pow__", seq, [42, -3, True, 0], [seq_int, seq_bool, -seq_int])
         _test_binary("__ipow__", seq, [42, -3, True, 0], [seq_int, seq_bool, -seq_int], inplace=True)
-        assert_raises(ValueError, _test_binary, "__pow__", seq_int, [-3], [])
-        assert_raises(ValueError, _test_binary, "__ipow__", seq_int, [-3], [], inplace=True)
-
+        
+        with pytest.raises(ValueError):
+            _test_binary("__pow__", seq_int, [-3], [])
+            _test_binary("__ipow__", seq_int, [-3], [], inplace=True)
+        
         # __itruediv__ is only valid with float arrseq.
         for scalar in SCALARS + ARRSEQS:
-            assert_raises(TypeError, getattr(seq_int.copy(), "__itruediv__"), scalar)
+            with pytest.raises(TypeError):
+                seq_int_cp = seq_int.copy()
```

Comment:
Let's place this line outside the `with` context, so that the lines that can raise the error are minimal.